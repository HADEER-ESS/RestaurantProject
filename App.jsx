import React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
// const createNativeStackNavigator = require('@react-navigation/native-stack');
import {NavigationContainer} from '@react-navigation/native';
// const NavigationContainer = require('@react-navigation/native');
import {QueryClientProvider, QueryClient} from 'react-query';
// const QueryClientProvider = require('react-query');
// const QueryClient = require('react-query');
import HomeScreen from './src/screens/HomeScreen';
// const HomeScreen = require('./src/screens/HomeScreen');
import DetailsScreen from './src/screens/DetailsScreen';
// const DetailsScreen = require('./src/screens/DetailsScreen');

const Stack = createNativeStackNavigator();

const App = () => {
  const client = new QueryClient();
  return (
    <QueryClientProvider client={client}>
      <NavigationContainer>
        <Stack.Navigator screenOptions={{headerShown: false}}>
          <Stack.Screen name="Home" component={HomeScreen} />
          <Stack.Screen name="Details" component={DetailsScreen} />
        </Stack.Navigator>
      </NavigationContainer>
    </QueryClientProvider>
  );
};

export default App;
