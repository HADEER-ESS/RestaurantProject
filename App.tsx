/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import {QueryClientProvider, QueryClient} from 'react-query';
import HomeScreen from './src/screens/HomeScreen';
import DetailsScreen from './src/screens/DetailsScreen';


const Stack = createNativeStackNavigator();

function App() {
  const client = new QueryClient() 
  return (
    <QueryClientProvider client={client}>
    <NavigationContainer>
      <Stack.Navigator screenOptions={{headerShown : false}}>
        <Stack.Screen name='Home' component={HomeScreen}/>
        <Stack.Screen name='Details' component={DetailsScreen}/>
      </Stack.Navigator>
    </NavigationContainer>
    </QueryClientProvider>
  );
}


export default App;
