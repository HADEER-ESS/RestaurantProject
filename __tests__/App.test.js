import 'react-native';
import React from 'react';
import App from '../App.jsx';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// beforeEach(() => {
//   jest.useFakeTimers();
// });

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
  // renderer.create(<App />);
});
