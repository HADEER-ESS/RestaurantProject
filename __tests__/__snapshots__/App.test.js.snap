// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 17,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "#000",
                    "borderRadius": 7,
                    "opacity": 1,
                    "padding": 9,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#ffffff",
                        "fontSize": 19,
                      },
                      undefined,
                      {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <Text
                style={
                  {
                    "color": "#000000",
                    "fontSize": 25,
                    "fontWeight": "700",
                    "paddingLeft": "25%",
                  }
                }
              >
                Restaurants
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#000000",
                  "borderRadius": 13,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "height": 37,
                  "justifyContent": "space-between",
                  "marginHorizontal": 17,
                  "marginVertical": 22,
                  "paddingHorizontal": 11,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#000000",
                      "fontSize": 19,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
              <TextInput
                onChangeText={[Function]}
                style={
                  {
                    "paddingVertical": 7,
                    "width": "80%",
                  }
                }
                value=""
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontSize": 19,
                      },
                      undefined,
                      {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <RCTScrollView>
              <View>
                <RCTSafeAreaView
                  style={
                    {
                      "paddingHorizontal": 7,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#000000",
                          "fontSize": 23,
                          "fontWeight": "700",
                        }
                      }
                    >
                      Cost Effective
                    </Text>
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontSize": 23,
                            },
                            undefined,
                            {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                  <ActivityIndicator
                    color="#000"
                    size="large"
                  />
                  <Text
                    style={
                      {
                        "color": "#000000",
                        "fontSize": 23,
                        "fontWeight": "700",
                      }
                    }
                  >
                    Bit Price
                  </Text>
                  <ActivityIndicator
                    color="#000"
                    size="large"
                  />
                  <Text
                    style={
                      {
                        "color": "#000000",
                        "fontSize": 23,
                        "fontWeight": "700",
                      }
                    }
                  >
                    Bi Spender
                  </Text>
                  <ActivityIndicator
                    color="#000"
                    size="large"
                  />
                </RCTSafeAreaView>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
